deepspeed/runtime/engine.py

```diff
@@ -3319,6 +3320,7 @@ def _save_checkpoint(self, save_dir, tag, client_state={}, exclude_frozen_parame
                      ds_config=self.config,
                      ds_version=version)
         state.update(client_state)
+        inject_universal_info(state)
```

Comment:
Shoudn't we show a warning when we don't have the necessary info?
It will silently produce an incorrect checkpoint if the checkpoint is loaded for TP or PP.
We can say that the converted checkpoint is only for pure DP scaling.

Reply:
After I discussed with Tunji, we are considering another approach for this. He will share a new approach. I keep this comment but just disregard it.

Reply:
@xylian86, the new approach is to do the injection in the conversion script rather than during saving. The injection should be done into `ds_checkpoint` before this [assertion](https://github.com/microsoft/DeepSpeed/blob/ebf82e8f3ad6d51d49d115e54a11ae4597ff36fb/deepspeed/checkpoint/ds_to_universal.py#L358).  Furthermore, the injection should be enabled by command-line argument (disabled by default) so that users are fully aware of what is going on. The command-line arg could be called `--inject-missing-state`.